pints/_mcmc/__init__.py

```diff
@@ -300,22 +309,65 @@ class MCMCController(object):
     """
 
     def __init__(
-            self, log_pdf, chains, x0, sigma0=None, transform=None,
+            self, log_pdf, chains, x0=None, sigma0=None, transform=None,
             method=None):
 
         # Check function
         if not isinstance(log_pdf, pints.LogPDF):
             raise ValueError('Given function must extend pints.LogPDF')
 
+        # Get number of parameters
+        self._n_parameters = log_pdf.n_parameters()
+
+        # Check number of chains
+        self._n_chains = int(chains)
+        if self._n_chains < 1:
+            raise ValueError('Number of chains must be at least 1.')
+
+        # Get functions which can generate initial position or set initial
+        # position points if numerical values given
+        self._x0_isfunction = False
+        if x0 is None:
+            # use prior if one's available
+            if isinstance(log_pdf, pints.LogPosterior):
+                x0 = log_pdf.log_prior()
+            # do what Stan does and initialise all parameters on U(-2, 2) scale
```

Comment:
I'd prefer raising a ValueError at this point, saying users must specify an x0 or provide a log posterior. I've had some experience with people leaving things at their default values and then running into problems, so better to make them aware there's an important choice to be made

Reply:
I would prefer to keep this as is: this is the default used by the major probabilistic programming languages, so think it's not really unsafe (what's more unsafe is allowing a user to specify a fixed x0 but we need that).